vault/eventbus/bus.go

```diff
@@ -103,7 +149,23 @@ func NewEventBus(logger hclog.Logger) (*EventBus, error) {
 	}, nil
 }
 
-func (bus *EventBus) Subscribe(_ context.Context, eventType logical.EventType) (chan *logical.EventData, error) {
+func nsEventTypeJoin(ns *namespace.Namespace, eventType logical.EventType) eventlogger.EventType {
+	if ns == nil {
+		ns = namespace.RootNamespace
+	}
+	return eventlogger.EventType(namespace.Canonicalize(ns.Path) + string(eventType))
+}
+
+func nsEventTypeSplit(eventType eventlogger.EventType) (string, logical.EventType) {
```

Comment:
I don't love this, as there is bound to be confusion with paths, namespaces, or slashes in the plugin event types. Maybe we should `hex` or `base64` encode each piece so we can guarantee the pieces can't conflict?